src/track/beats.cpp

```diff
@@ -60,23 +60,23 @@ Beats::ConstIterator Beats::ConstIterator::operator+=(Beats::ConstIterator::diff
     }
 
     DEBUG_ASSERT(n > 0);
-    const int beatOffset = m_beatOffset + n;
 #ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
     const auto origValue = m_value;
 #endif
 
-    // Detect integer overflow
-    if (beatOffset < m_beatOffset) {
+    // Detect integer overflow in `m_beatOffset + n`
+    const int maxBeatOffset = std::numeric_limits<Beats::ConstIterator::difference_type>::max();
+    if (m_beatOffset > maxBeatOffset - n) {
         qWarning() << "Beats: Iterator would go out of possible range, capping "
```

Comment:
I am trying to find an explanation why the original overflow condition wouldn't work. At a first glance, the result should be the same. That said, this is certainly less obscure and more readable. Still, it would be nice to understand the difference, and I don't see that yet.

Reply:
(My understanding, might be wrong) is that the compiler is allowed to assume that signed overflow never happens and may optimize away the entire if-statement to a noop. In any case, we shouldn't rely on it.